cmake/today.cmake

```diff
@@ -3,14 +3,10 @@
 # YYYY-MM-DD
 #
 MACRO (TODAY RESULT)
-    IF (WIN32)
-        EXECUTE_PROCESS(COMMAND "cmd" " /C date +%Y-%m-%d" OUTPUT_VARIABLE ${RESULT})
-        string(REGEX REPLACE "(..)/(..)/..(..).*" "\\1/\\2/\\3" ${RESULT} ${${RESULT}})
-    ELSEIF(UNIX)
-        EXECUTE_PROCESS(COMMAND "date" "+%Y-%m-%d" OUTPUT_VARIABLE ${RESULT})
-        string(REGEX REPLACE "(..)/(..)/..(..).*" "\\1/\\2/\\3" ${RESULT} ${${RESULT}})
-    ELSE (WIN32)
-        MESSAGE(SEND_ERROR "date not implemented")
-        SET(${RESULT} 000000)
-    ENDIF (WIN32)
+    if (DEFINED ENV{SOURCE_DATE_EPOCH})
+        EXECUTE_PROCESS(COMMAND "date" "-u" "-d" "@$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d" 
+                        COMMAND "tr" "-d" "'\n'" OUTPUT_VARIABLE ${RESULT})
```

Comment:
`date` is probably only available on UNIX and not WIN32
and https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal#CMake suggests to use OUTPUT_STRIP_TRAILING_WHITESPACE option